integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -821,72 +822,78 @@ object CarbonDataRDDFactory {
+              Some(carbonTable.getDatabaseName))), None)
+        var storeLocation = ""
+        val configuredStore = Util.getConfiguredLocalDirs(SparkEnv.get.conf)
+        if (null != configuredStore && configuredStore.nonEmpty) {
+          storeLocation = configuredStore(Random.nextInt(configuredStore.length))
+        }
+        if (storeLocation == null) {
+          storeLocation = System.getProperty("java.io.tmpdir")
+        }
+        storeLocation = storeLocation + "/carbonstore/" + System.nanoTime()
 
-        if (lock.lockWithRetries()) {
-          LOGGER.info("Acquired the compaction lock.")
-          try {
-            startCompactionThreads(sqlContext,
-              carbonLoadModel,
-              storeLocation,
-              compactionModel,
-              lock,
-              compactedSegments,
-              operationContext
-            )
-          } catch {
-            case e: Exception =>
-              LOGGER.error(s"Exception in start compaction thread. ${ e.getMessage }")
-              lock.unlock()
-              throw e
-          }
+        val isConcurrentCompactionAllowed = CarbonProperties.getInstance().getProperty(
+          CarbonCommonConstants.ENABLE_CONCURRENT_COMPACTION,
+          CarbonCommonConstants.DEFAULT_ENABLE_CONCURRENT_COMPACTION
+        ).equalsIgnoreCase("true")
+
+        if (!isConcurrentCompactionAllowed) {
+          handleCompactionForSystemLocking(sqlContext,
+            carbonLoadModel,
+            storeLocation,
+            CompactionType.MINOR,
+            carbonTable,
+            compactedSegments,
+            compactionModel,
+            operationContext
+          )
         } else {
-          LOGGER.error("Not able to acquire the compaction lock for table " +
-                       s"${ carbonLoadModel.getDatabaseName }.${ carbonLoadModel.getTableName}")
+          val lock = CarbonLockFactory.getCarbonLockObj(
+            carbonTable.getAbsoluteTableIdentifier,
+            LockUsage.COMPACTION_LOCK)
+
+          if (lock.lockWithRetries()) {
+            LOGGER.info("Acquired the compaction lock.")
+            try {
+              startCompactionThreads(sqlContext,
+                carbonLoadModel,
+                storeLocation,
+                compactionModel,
+                lock,
+                compactedSegments,
+                operationContext
+              )
+            } catch {
+              case e: Exception =>
+                LOGGER.error(s"Exception in start compaction thread. ${ e.getMessage }")
+                lock.unlock()
+                throw e
+            }
+          } else {
+            LOGGER.error("Not able to acquire the compaction lock for table " +
+                         s"${ carbonLoadModel.getDatabaseName }.${ carbonLoadModel.getTableName }")
+          }
         }
       }
+    } catch {
+      case ex: Exception =>
+        LOGGER.error(s"Auto Compaction failed for table ${
```

Comment:
Mentioned that compaction got failed but ignoring.

Reply:
ok